### PR TITLE
[Core] add 'recently viewed characters' && add character link to report

### PR DESCRIPTION
**Pull request by joshinat0r**
Adds recently viewed characters to the 'recently viewed logs' div. (#1965 )
Pretty sure this:
```
const mapStateToProps = state => {
  return ({ });
};
 export default withRouter(connect(
  mapStateToProps,
  {
    appendReportHistory,
  }
)(Parses));
```
isn't how it's supposed to be done, but it doesn't work without mapStateToProps.

image

Also adds a link to our character-view to the report page if we have the realm & region info for that character.
image

### CODE REVIEW

**Comment by joshinat0r** [comment]
**Don't merge**

Just realized that the realm-field has translated names (as in: our API saves for the Korean realm `Azshara` Azshara, but the character-URL want's the translated version 아즈샤라).

**Comment by joshinat0r** [comment]
Removed the link to our character-page from the report-result-page.
Implementing this would either mean 
- that our API has save the slug or the untranslated name 
- that we add something like a `originalName` to the `REALMS.js` (but the export from the bnet-API doesn't include those)

**Comment by joshinat0r** [comment]
- Readded character link to the report page
- Updated our character API to respect the encoding for each region

File: src/Interface/Character/Parses.js [hunk 1]
```diff
@@ -544,4 +570,13 @@ class Parses extends React.Component {
   }
 }
 
-export default Parses;
+const mapStateToProps = state => {
+  return ({ });
+};
+
+export default withRouter(connect(
+  mapStateToProps,
```
Comment (MartijnHols) [inline src/Interface/Character/Parses.js]: Since this is unused this should be null

File: src/Interface/Character/Parses.js [hunk 2]
```diff
@@ -544,4 +570,13 @@ class Parses extends React.Component {
   }
 }
 
-export default Parses;
+const mapStateToProps = state => {
+  return ({ });
+};
+
+export default withRouter(connect(
```
Comment (MartijnHols) [inline src/Interface/Character/Parses.js]: When using multiple decorators please use the `compose` function from the `redux` library.

File: src/Interface/Others/Report/Results/index.js [hunk 3]
```diff
@@ -279,6 +279,17 @@ class Results extends React.PureComponent {
...
               </a>
+              {' '}
+              {characterProfile && characterProfile.realm && characterProfile.name && characterProfile.region && (
+                <Link 
+                  to={`/character/${characterProfile.region.toUpperCase()}/${characterProfile.realm}/${characterProfile.name}/`} 
+                  data-tip={`Show ${characterProfile.realm} - ${characterProfile.name}'s most recent logs`}
+                  className="btn"
+                  style={{ fontSize: 24 }}
+                >
+                  <img src="/favicon.png" alt="Wipefest logo" style={{ height: '1.4em', marginTop: '-0.15em' }} /> {characterProfile.name}
```
Comment (MartijnHols) [inline src/Interface/Others/Report/Results/index.js]: "Wipefest logo"

File: src/Interface/Others/Report/Results/index.js [hunk 4]
```diff
@@ -279,6 +279,17 @@ class Results extends React.PureComponent {
               >
                 <img src={WipefestLogo} alt="Wipefest logo" style={{ height: '1.4em', marginTop: '-0.15em' }} /> Wipefest
               </a>
+              {' '}
+              {characterProfile && characterProfile.realm && characterProfile.name && characterProfile.region && (
+                <Link 
+                  to={`/character/${characterProfile.region.toUpperCase()}/${characterProfile.realm}/${characterProfile.name}/`} 
+                  data-tip={`Show ${characterProfile.realm} - ${characterProfile.name}'s most recent logs`}
```
Comment (MartijnHols) [inline src/Interface/Others/Report/Results/index.js]: Show -> View (view is what the users wants to do, show is what the system would do)

logs -> reports (we call them reports usually in line with WCL's naming)

**Comment by joshinat0r** [comment]
Fixed the naming and updated the export to use compose

File: server/helpers/wowCommunityApi.js [hunk 5]
```diff
@@ -1,20 +1,20 @@
...
+  'eu': 'ru_RU',
+  'us': 'en_US',
+  'tw': 'zh_TW',
+  'kr': 'ko_KR',
+};
 
 export function fetchCharacter(region, realm, name, fields) {
   region = region.toLowerCase();
-  if (!availableRegions.includes(region)) {
+  if (!Object.keys(availableRegions).includes(region)) {
```
Comment (MartijnHols) [inline server/helpers/wowCommunityApi.js]: This could be just `availableRegions[region]`

File: src/Interface/Character/Parses.js [hunk 6]
```diff
@@ -13,9 +16,11 @@ import ZONES from 'common/ZONES';
 import SPECS from 'common/SPECS';
 import DIFFICULTIES from 'common/DIFFICULTIES';
 import ITEMS from 'common/ITEMS';
+import REPORT_HISTORY_TYPES from 'Interface/Home/ReportHistory/REPORT_HISTORY_TYPES';
 
 import './Parses.css';
 import ParsesList from './ParsesList';
+import { compose } from '../../../node_modules/redux';
```
Comment (MartijnHols) [inline src/Interface/Character/Parses.js]: Please change this to an absolute import

File: src/Interface/Character/Parses.js [hunk 7]
```diff
@@ -1,6 +1,9 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { Link } from 'react-router-dom';
+import { appendReportHistory } from 'Interface/actions/reportHistory';
```
Comment (MartijnHols) [inline src/Interface/Character/Parses.js]: Please put project imports in a separate group from library imports.

**Comment by joshinat0r** [comment]
Fixed

**Pull request merged**